code/_onclick/hud/alert.dm

```diff
@@ -256,6 +258,74 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
 	icon_state = "blobbernaut_nofactory"
 	alerttooltipstyle = "blob"
 
+// BLOODCULT
+
+/obj/screen/alert/bloodsense
+	name = "Blood Sense"
+	desc = "Allows you to sense blood that is manipulated by dark magicks."
+	icon_state = "cult_sense"
+	var/image/finder
+	var/image/sacimage
+	var/mob/sacmob
+
+/obj/screen/alert/bloodsense/Initialize()
+	..()
+	finder = new('icons/mob/screen_alert.dmi', "no_sacrifice")
+	var/datum/mind/sacmind = GLOB.sac_target
+	sacmob = sacmind.current
+	var/datum/job/sacjob = SSjob.GetJob(sacmind.assigned_role)
+	var/datum/preferences/sacpref = sacmob.client.prefs
+	var/icon/reshape = get_flat_human_icon(null, sacjob, sacpref)
+	reshape.Shift(SOUTH, 4)
+	reshape.Crop(7,4,26,30)
+	reshape.Crop(-5,-2,26,29)
+	sacimage = reshape
+	add_overlay(finder)
+	START_PROCESSING(SSprocessing, src)
+	process()
+
+/obj/screen/alert/bloodsense/Destroy()
+    qdel(finder)
+    finder = null
+    STOP_PROCESSING(SSprocessing, src)
+    return ..()
+
+/obj/screen/alert/bloodsense/process()
+	if (!GLOB.sac_target)
+		return
+	if(!GLOB.blood_target && !GLOB.sac_complete)
+		cut_overlays()
+		finder.icon_state = "no_sacrifice"
+		desc = "Nar-Sie demands that [GLOB.sac_target] be sacrificed before the summoning can begin."
+		add_overlay(finder)
+		add_overlay(sacimage)
+		return
+	if(!GLOB.blood_target && GLOB.sac_complete)
+		cut_overlays()
+		finder.icon_state = "sacrifice"
+		desc = "The sacrifice is complete, prepare to summon Nar-Sie!"
+		add_overlay(finder)
+		return
+	var/turf/Q = get_turf(GLOB.blood_target)
+	var/turf/A = get_turf(mob_viewer)
+	if(Q.z != A.z) //The target is on a different Z level, we cannot sense that far.
```

Comment:
I could add a check for the latter but if it exists wouldn't it always have a turf?